src/main/java/elements/MyList.java

```diff
@@ -0,0 +1,12 @@
+package elements;
+
+import org.openqa.selenium.WebElement;
+
+import java.util.Collection;
+import java.util.List;
+
+public interface MyList<P extends WebElement> extends Collection<PageElement> {
```

Comment:
I don't think that this interface should be parametrised
Remove <P extends WebElement>
We will use only PageElement